src/Maestro/tests/Maestro.ScenarioTests/Shareable.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading;
+
+namespace Maestro.ScenarioTests
+{
+    public sealed class Shareable<T> : IDisposable where T : class, IDisposable
+    {
+        private T? _inner;
+
+        internal Shareable(T inner)
+        {
+            _inner = inner;
+        }
+
+        public void Dispose()
+        {
+            _inner?.Dispose();
+        }
+
+        public T? TryTake()
+        {
+            return Interlocked.Exchange(ref _inner, null);
+        }
+
+        public T? Peek()
+        {
+            return _inner;
```

Comment:
It'd probably make "Peek" throw if it's null.  You shouldnt' be peeking something that was taken.

Reply:
Done.